app/controllers/boxes_controller.rb

```diff
@@ -100,12 +113,37 @@ def load_box
     end
   end
 
+  def load_box_print
+
+    @box = Box.where(institution: @navigation_context.institution, id: params.fetch(:id)).take
+    if @boxes.nil?
+      @box = Box
+        .joins(box_transfers: :transfer_package)
+        .where("boxes.id = ? and ( transfer_packages.sender_institution_id = ? or transfer_packages.receiver_institution_id = ? )", params.fetch(:id), @navigation_context.institution, @navigation_context.institution).take
+    end
+  end
+
   def load_box_samples
     samples = @box.samples.preload(:batch, :sample_identifiers)
-    samples = samples.scrambled if @box.blinded?
     SamplePresenter.map(samples, request.format)
   end
 
+  def load_box_samples_print
+    samples = @box.samples.preload(:batch, :sample_identifiers)
+    if !@box.blinded or is_sender
+      samples = samples.sort_by{|sample| [sample.batch_number, sample.concentration, sample.replicate ]}
+    else
+      samples = samples.scrambled #this sorts by uuid
+    end
+  end
+
+  def is_sender
+    return BoxTransfer
+      .joins(:transfer_package)
+      .where( box_id: params.fetch(:id), transfer_packages: { sender_institution_id: @navigation_context.institution } )
+      .any?
```

Comment:
In Ruby `.any?` isn't the opposite of `.empty?`. It usually involves iterating the collection for a truthy value (similar to `Array.prototype.some` in JavaScript), while `.empty?` merely checks that it's size is zero.

I'm not sure what ActiveRecord does, but the fastest method is `.exists?` (no iteration, no loading, not even a SQL count):

```suggestion
      .exists?
```

You'll may want to memoize the result, so you can repeatedly call the method. We also prefer a question mark suffix to is/has prefixes in Ruby:

```ruby
def sender?
  return @sender unless @sender.nil?
  @sender = BoxTransfer...
end
```

Reply:
It was `.present?` @straight-shoota suggested `.any?`. In [this link](https://andreigridnev.com/blog/2016-04-17-empty-blank-any-exists-methods-of-ruby-on-rails-activerecord/) says that the fastest is `.exists`. So I will take your suggestion. :)

Reply:
Ah, yes `exists?` is indeed better here. But that's just because of the old Rails version. Only in newer versions what I said above is true and `any?` is equivalent to `exists?`.